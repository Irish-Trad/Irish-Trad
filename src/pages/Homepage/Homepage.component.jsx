import React from 'react'

import {
  MainContainer,
  H1,
  Figure,
  Img,
  Figcaption,
  H2,
  Text,
  ExternalLink
} from './Homepage.styles'

const Homepage = () => (
  <MainContainer>
    <H1>Irish-Trad.net</H1>
    <Text>
      Irish-Trad.net provides free online resources and lessons for a variety of
      stringed instruments commonly found in traditional Irish sessions such as
      the Irish Bouzouki, Mandolin, Octave Mandola and Irish Tenor Banjo.
    </Text>
    <Figure>
      <Img
        src="../assets/low-quality/2243E.jpg"
        alt="Ozark 2243E Irish Bouzouki"
        title="Ozark 2243E Irish Bouzouki"
      />
      <Figcaption>Pictured: Ozark 2243E Irish Bouzouki</Figcaption>
    </Figure>
    <Text>
      As these instruments can be tuned to a standard GDAE or alternative GDAD
      tuning all resources that showcase chord diagrams or scales will apply.
    </Text>
    <Text>
      The Irish Bouzouki is a popular stringed instrument used for accompaniment
      during traditional Irish sessions. Due to the Irish Bouzouki's scale
      length and overall pitch this tends to be in the form of chordal
      accompaniment however the instrument is also capable of performing melody
      lines. It is a relatively young instrument and a lot of its history &amp;
      origin can be found on Wikipedia:{' '}
      <ExternalLink
        href="https://en.wikipedia.org/wiki/Irish_Bouzouki"
        target="_blank"
      >
        Wikipedia - Irish Bouzouki
      </ExternalLink>
      .
    </Text>
    <Text>
      Most people who discover the Irish Bouzouki for the first time do so
      having already played a different stringed instrument such as the guitar,
      mandolin or banjo. For guitar players, the most popular method of teaching
      is through tablature because of its simplicity and ability to get a person
      playing quickly. The lessons and resources on Irish-Bouzouki.net are
      presented in tablature (TAB) alongside traditional sheet music to satisfy
      the absolute beginner and the musically trained.
    </Text>
    <Text>
      Irish-Bouzouki.net will not be hosting any copyrighted tunes or songs but
      instead provides all the information and musical theory required to bring
      a player to the level where they can use a website such as{' '}
      <ExternalLink href="https://thesession.org/">The Session</ExternalLink> to
      play any of the tunes available there.
    </Text>
    <H2>Lessons</H2>
    <Text>
      The lessons section will provide information such as open chords, basic
      scales, barre and 'moveable' chords and additional information such as how
      to use accessories like a capo or how to change and tune Bouzouki strings.
      Music theory will also be used to reinforce an understanding of how chords
      and scales are construced.
    </Text>
    <H2>Chords</H2>
    <Text>
      The chords section provides chord charts for the two most common tunings -
      GDAE and GDAD. The section covers basic triads such as Major and Minor and
      extended chords such as Major 7th, Minor 7th and Dominant 7th. The section
      will be expanded to include 'quick' moveable chords.
    </Text>
    <H2>Clef to TAB</H2>
    <Text>
      The Cleft-to-TAB section provides a conversion chart that allows for any
      piece of sheet music found in a book or online to be converted into
      tablature - as long as the piece music covers less than 4 octaves (the
      range of the Irish Bouzouki).
    </Text>
    <H2>Terminology</H2>
    <Text>
      The terminology section provides an A-Z of common phrases, names and
      abbreviations that can be found across traditional irish music and session
      settings.
    </Text>
  </MainContainer>
)

export default Homepage
