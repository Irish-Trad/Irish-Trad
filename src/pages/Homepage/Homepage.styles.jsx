import styled from 'styled-components'

export const Container = styled.main`
  max-width: 960px;
  margin: auto;
  padding: 10px;
`

export const H1 = styled.h1`
  text-align: center;
  font-family: meathflfregular;
  font-size: 55px;
  margin: 10px;
`

export const H2 = styled.h2`
  font-size: 24px;
  font-weight: bold;
  margin: 10px 0px;
`

export const Text = styled.p``

export const Figure = styled.figure`
  display: flex;
  flex-direction: column;
  justify-content: ;
  align-items: center;
`

export const Img = styled.img`
  width: 50%;

  @media screen and (min-width: 750px) {
    width: 25%;
  }
`

export const Figcaption = styled.figcaption``

export const Link = styled.a`
  text-decoration: none;
  color: black;
  cursor: pointer;
`
