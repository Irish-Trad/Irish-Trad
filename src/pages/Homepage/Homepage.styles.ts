import styled from 'styled-components'
import Image from './image/session.jpg'

export const ExternalLink = styled.a`
  text-decoration: none;
  color: black;
  cursor: pointer;
`

export const H1 = styled.h1`
  text-align: center;
  font-family: meathflfregular;
  font-size: 10em;
  margin: 10px;
`

export const IntroContainer = styled.div`
  background-image: url(${Image});
  background-size: cover;
  display: flex;
  justify-content: center;
  align-items: center;
  color: white;
  flex-flow: column nowrap;
  padding: 50px 200px;

  @media screen and (min-width: 750px) {
    height: 100vh;
  }
`

export const H2 = styled.h2`
  font-size: 2em;
  font-weight: bold;
  margin: 10px 0px;
  text-align: center;
  text-decoration: underline green 5px;
  text-underline-offset: 15px;
`

export const Text = styled.p`
  font-size: 32px;
`

export const Divider = styled.section`
  box-shadow: inset 0 5px 0 0 rgba(0, 0, 0, 0.05),
    inset 0 0.1em 0.1em 0 rgba(0, 0, 0, 0.025);

  @media screen and (min-width: 750px) {
    padding: 50px 200px;
  }
`
