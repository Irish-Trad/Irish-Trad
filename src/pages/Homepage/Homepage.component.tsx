import React from 'react'
import Metadata from 'common/Metadata'

import {
  H1,
  Figure,
  Image,
  Figcaption,
  H2,
  Text,
} from 'common/StyledComponents/StyledComponents'

import { ExternalLink } from './Homepage.styles'

const Homepage: React.FC = () => (
  <>
    <Metadata
      title="Irish-Trad.net - Home"
      metaTitle="Irish-Trad.net - Home"
      metaKeywords="Irish, Irish Trad, Irish Traditional Music, Irish Folk Music, Irish Bouzouki, Bouzouki, Mandolin, Octave Mandolin, Mandola, Octave Mandola, Chords, TAB, Tablature"
      metaDescription="Free Online Lessons and Resources for Irish Bouzouki/Mandolin/Octave Mandola/Irish Tenor Banjo"
    />
    <H1>Irish-Trad.net</H1>
    <Text>
      Irish-Trad.net provides free online resources and lessons for a variety of
      stringed instruments commonly found in traditional Irish sessions such as
      the Irish Bouzouki, Mandolin, Octave Mandola and Irish Tenor Banjo.
    </Text>
    <Figure>
      <Image
        src="assets/low-quality/2243E.jpg"
        alt="Ozark 2243E Irish Bouzouki"
        title="Ozark 2243E Irish Bouzouki"
      />
      <Figcaption>Pictured: Ozark 2243E Irish Bouzouki</Figcaption>
    </Figure>
    <Text>
      Since all of these instruments can be tuned to the standard GDAE or
      alternative GDAD tuning all of the resources found on Irish-Trad.net such
      as chord diagrams and scales will apply.
    </Text>
    <Text>
      The Irish Bouzouki is a popular stringed instrument used for accompaniment
      during traditional Irish sessions. Due to the Irish Bouzouki's scale
      length and overall pitch this tends to be in the form of chordal
      accompaniment however the instrument is also capable of performing melody
      lines. It is a relatively young instrument and a lot of its history &amp;
      origin can be found on Wikipedia:{' '}
      <ExternalLink
        href="https://en.wikipedia.org/wiki/Irish_Bouzouki"
        target="_blank"
      >
        Wikipedia - Irish Bouzouki
      </ExternalLink>
      .
    </Text>
    <Text>
      Most people who discover the Irish Bouzouki for the first time do so
      having already played a different stringed instrument such as the guitar,
      mandolin or banjo. For guitar players, the most popular method of teaching
      is through tablature because of its simplicity and ability to get a person
      playing quickly. The lessons and resources on Irish-Trad.net are presented
      in tablature (TAB) alongside traditional sheet music to satisfy the
      absolute beginner and the musically trained.
    </Text>
    <Text>
      Irish-Trad.net will not be hosting any copyrighted tunes or songs but
      instead provides all the information and musical theory required to bring
      a player to the level where they can use a website such as{' '}
      <ExternalLink href="https://thesession.org/">The Session</ExternalLink> to
      play any of the tunes available there.
    </Text>
    <H2>Lessons</H2>
    <Text>
      The lessons section will provide information such as open chords, basic
      scales, barre and 'moveable' chords and additional information such as how
      to use accessories like a capo or how to change and tune Bouzouki strings.
      Music theory will also be used to reinforce an understanding of how chords
      and scales are construced.
    </Text>
    <H2>Chords</H2>
    <Text>
      The chords section provides chord charts for the two most common tunings -
      GDAE and GDAD. The section covers basic triads such as Major and Minor and
      extended chords such as Major 7th, Minor 7th and Dominant 7th. The section
      will be expanded to include 'quick' moveable chords.
    </Text>
    <H2>Clef to TAB</H2>
    <Text>
      The Cleft-to-TAB section provides a conversion chart that allows for any
      piece of sheet music found in a book or online to be converted into
      tablature - as long as the piece music covers less than 4 octaves (the
      range of the Irish Bouzouki).
    </Text>
    <H2>Terminology</H2>
    <Text>
      The terminology section provides an A-Z of common phrases, names and
      abbreviations that can be found across traditional irish music and session
      settings.
    </Text>
  </>
)

export default Homepage
