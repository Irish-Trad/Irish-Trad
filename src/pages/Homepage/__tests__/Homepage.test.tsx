import { render, cleanup } from '@testing-library/react'
import Homepage from '../Homepage.component'

const renderComponent = () => render(<Homepage />)

describe('Homepage', () => {
  beforeEach(() => {
    document.body.innerHTML = "<div id='top'></div>"
  })

  afterEach(cleanup)

  it('Snapshot', () => {
    const snapshot = renderComponent()

    expect(snapshot).toMatchSnapshot()
  })

  it('Title of the page is Irish-Trad.net', () => {
    const { getByText } = renderComponent()

    expect(getByText('Irish-Trad.net')).toBeInTheDocument()
  })
})
