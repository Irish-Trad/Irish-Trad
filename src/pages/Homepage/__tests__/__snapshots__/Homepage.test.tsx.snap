// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxiKw dGVAop"
      >
        <h1
          class="sc-AxirZ hbBtUA"
        >
          Irish-Trad.net
        </h1>
        <p
          class="sc-AxhUy hETkew"
        >
          Irish-Trad.net provides free online resources and lessons for a variety of stringed instruments commonly found in traditional Irish sessions such as the Irish Bouzouki, Mandolin, Octave Mandola and Irish Tenor Banjo.
        </p>
      </div>
      <section
        class="sc-AxgMl bJJhx"
      >
        <h2
          class="sc-AxhCb iCBRgr"
        >
          Introduction
        </h2>
        <p
          class="sc-AxhUy hETkew"
        >
          The resources found on Irish-Trad.net such as chord diagrams and scales refer to any instrument that can be tuned to the standard GDAE or alternative GDAD tuning.
        </p>
        <p
          class="sc-AxhUy hETkew"
        >
          Most people who decide to learn traditional Irish music for the first time do so having already played a different stringed instrument such as the guitar. For guitar players, the most popular method of teaching is through tablature because of the simplicity and ability to get a person playing quickly. The lessons and resources on Irish-Trad.net are presented in tablature (TAB) alongside traditional sheet music to satisfy the absolute beginner and the musically trained.
        </p>
        <p
          class="sc-AxhUy hETkew"
        >
          Irish-Trad.net will not be hosting any copyrighted tunes or songs and instead aims to provide all the information and musical theory required to bring a player to the level where they can use a website such as
          <a
            class="sc-AxjAm iItssD"
            href="https://thesession.org/"
          >
             The Session
          </a>
          to play any of the tunes available there.
        </p>
      </section>
      <section
        class="sc-AxgMl bJJhx"
      >
        <h2
          class="sc-AxhCb iCBRgr"
        >
          Lessons
        </h2>
        <p
          class="sc-AxhUy hETkew"
        >
          The lessons section will provide information such as open chords, basic scales, moveable chords (barre) and additional information such as how to use accessories like a capo or how to change and tune strings. Music theory will also be used to reinforce an understanding of how chords and scales are construced.
        </p>
      </section>
      <section
        class="sc-AxgMl bJJhx"
      >
        <h2
          class="sc-AxhCb iCBRgr"
        >
          Chords
        </h2>
        <p
          class="sc-AxhUy hETkew"
        >
          The chords section provides chord charts for the two most common tunings - GDAE and GDAD. The section covers basic triads such as Major and Minor and extended chords such as Major 7th, Minor 7th and Dominant 7th. The section will be expanded to include 'quick' moveable chords.
        </p>
      </section>
      <section
        class="sc-AxgMl bJJhx"
      >
        <h2
          class="sc-AxhCb iCBRgr"
        >
          Clef to TAB
        </h2>
        <p
          class="sc-AxhUy hETkew"
        >
          The Cleft-to-TAB section provides a conversion chart that allows for any piece of sheet music found in a book or online to be converted into tablature - as long as the piece music covers less than 4 octaves (the range of the Irish Bouzouki).
        </p>
      </section>
      <section
        class="sc-AxgMl bJJhx"
      >
        <h2
          class="sc-AxhCb iCBRgr"
        >
          Terminology
        </h2>
        <p
          class="sc-AxhUy hETkew"
        >
          The terminology section provides an A-Z of common phrases, names and abbreviations that can be found across traditional irish music and session settings.
        </p>
      </section>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxiKw dGVAop"
    >
      <h1
        class="sc-AxirZ hbBtUA"
      >
        Irish-Trad.net
      </h1>
      <p
        class="sc-AxhUy hETkew"
      >
        Irish-Trad.net provides free online resources and lessons for a variety of stringed instruments commonly found in traditional Irish sessions such as the Irish Bouzouki, Mandolin, Octave Mandola and Irish Tenor Banjo.
      </p>
    </div>
    <section
      class="sc-AxgMl bJJhx"
    >
      <h2
        class="sc-AxhCb iCBRgr"
      >
        Introduction
      </h2>
      <p
        class="sc-AxhUy hETkew"
      >
        The resources found on Irish-Trad.net such as chord diagrams and scales refer to any instrument that can be tuned to the standard GDAE or alternative GDAD tuning.
      </p>
      <p
        class="sc-AxhUy hETkew"
      >
        Most people who decide to learn traditional Irish music for the first time do so having already played a different stringed instrument such as the guitar. For guitar players, the most popular method of teaching is through tablature because of the simplicity and ability to get a person playing quickly. The lessons and resources on Irish-Trad.net are presented in tablature (TAB) alongside traditional sheet music to satisfy the absolute beginner and the musically trained.
      </p>
      <p
        class="sc-AxhUy hETkew"
      >
        Irish-Trad.net will not be hosting any copyrighted tunes or songs and instead aims to provide all the information and musical theory required to bring a player to the level where they can use a website such as
        <a
          class="sc-AxjAm iItssD"
          href="https://thesession.org/"
        >
           The Session
        </a>
        to play any of the tunes available there.
      </p>
    </section>
    <section
      class="sc-AxgMl bJJhx"
    >
      <h2
        class="sc-AxhCb iCBRgr"
      >
        Lessons
      </h2>
      <p
        class="sc-AxhUy hETkew"
      >
        The lessons section will provide information such as open chords, basic scales, moveable chords (barre) and additional information such as how to use accessories like a capo or how to change and tune strings. Music theory will also be used to reinforce an understanding of how chords and scales are construced.
      </p>
    </section>
    <section
      class="sc-AxgMl bJJhx"
    >
      <h2
        class="sc-AxhCb iCBRgr"
      >
        Chords
      </h2>
      <p
        class="sc-AxhUy hETkew"
      >
        The chords section provides chord charts for the two most common tunings - GDAE and GDAD. The section covers basic triads such as Major and Minor and extended chords such as Major 7th, Minor 7th and Dominant 7th. The section will be expanded to include 'quick' moveable chords.
      </p>
    </section>
    <section
      class="sc-AxgMl bJJhx"
    >
      <h2
        class="sc-AxhCb iCBRgr"
      >
        Clef to TAB
      </h2>
      <p
        class="sc-AxhUy hETkew"
      >
        The Cleft-to-TAB section provides a conversion chart that allows for any piece of sheet music found in a book or online to be converted into tablature - as long as the piece music covers less than 4 octaves (the range of the Irish Bouzouki).
      </p>
    </section>
    <section
      class="sc-AxgMl bJJhx"
    >
      <h2
        class="sc-AxhCb iCBRgr"
      >
        Terminology
      </h2>
      <p
        class="sc-AxhUy hETkew"
      >
        The terminology section provides an A-Z of common phrases, names and abbreviations that can be found across traditional irish music and session settings.
      </p>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
